src/install/fetch.ts

```diff
@@ -167,13 +167,25 @@ type FetchToStoreOptions = {
  */
 function fetchToStoreCached (opts: FetchToStoreOptions): Promise<void> {
   return memoize(opts.fetchLocks, opts.resolution.id, async function () {
-    if (opts.force || !await exists(opts.target)) {
-      await rimraf(opts.target)
+    const target = opts.target
+    const targetStage = `${opts.target}-stage`
```

Comment:
That might do it. The target folder will either be a version or a commit sha for git dependencies.